ravens/uml/visualize.py

```diff
@@ -75,141 +212,202 @@ def _parse_object_style(object_style_string: str):
+                                    d[a] = int(b)
+                            link_style[k] = d
+                else:
+                    # not strictly needed, we only read HDN/CX/CY keys below
+                    pass
+            except Exception:
+                pass
+
+            role_e = self._edge_role_by_cid.get(cid)
+            edge_hex = EDGE_MAP_CI.get(str(role_e).strip().casefold(), DEFAULT_EDGE_HEX) if role_e else DEFAULT_EDGE_HEX
+
+            links_data.append({
+                "source": str(s),
+                "target": str(t),
+                "type": str(c["Connector_Type"]).lower(),
+                "textStartTop": f"+{c['SourceRole']}" if pd.notna(c["SourceRole"]) else "",
                 "textStartTopHidden": link_style.get("LLT", {}).get("HDN", 0),
                 "textStartTopXPos": link_style.get("LLT", {}).get("CX", 0.0),
                 "textStartTopYPos": link_style.get("LLT", {}).get("CY", 0.0),
-                "textEndTop": f"+{connector.DestRole}" if not pd.isnull(connector.DestRole) else "",
+                "textEndTop": f"+{c['DestRole']}" if pd.notna(c["DestRole"]) else "",
                 "textEndTopHidden": link_style.get("LRT", {}).get("HDN", 0),
                 "textEndTopXPos": link_style.get("LRT", {}).get("CX", 0.0),
                 "textEndTopYPos": link_style.get("LRT", {}).get("CY", 0.0),
-                "textStartBtm": f"{connector.SourceCard}" if not pd.isnull(connector.SourceCard) else "",
+                "textStartBtm": f"{c['SourceCard']}" if pd.notna(c["SourceCard"]) else "",
                 "textStartBtmHidden": link_style.get("LLB", {}).get("HDN", 0),
                 "textStartBtmXPos": link_style.get("LLB", {}).get("CX", 0.0),
                 "textStartBtmYPos": link_style.get("LLB", {}).get("CY", 0.0),
-                "textEndBtm": f"{connector.DestCard}" if not pd.isnull(connector.DestCard) else "",
+                "textEndBtm": f"{c['DestCard']}" if pd.notna(c["DestCard"]) else "",
                 "textEndBtmHidden": link_style.get("LRB", {}).get("HDN", 0),
                 "textEndBtmXPos": link_style.get("LRB", {}).get("CX", 0.0),
                 "textEndBtmYPos": link_style.get("LRB", {}).get("CY", 0.0),
-                "color": _CIM_RGB_TO_HEX[line_color],
-            }
-
-            links_data.append(link_data)
+                "color": edge_hex,
+            })
 
         svg_data["links"] = links_data
-
         self._current_diagram = diagram_id
         self._current_svg_data = svg_data
 
     def _create_svg(self):
-        data_json = json.dumps(self._current_svg_data)
-
-        # Call the Node.js script
-        result = subprocess.run(["node", _SVG_RENDERER_PATH, data_json], capture_output=True, text=True)
-
-        if result.stderr:
-            raise Exception(result.stderr)
-
-        # Get the SVG output
-        self._current_svg = result.stdout
+        # write the (possibly large) payload to a temp file instead of passing on CLI
```

Comment:
I implemented the tmpfile because--at least on my system--streaming `data_json` through `subprocess` was _really_ slow. The tempfile just serializes the diagram specs to a temporary JSON file, then invokes Node with that filepath, reads back the SVG, then gets deleted. I can take this out if you want, but I think it makes RAVENS more platform-agnostic as OS buffer sizes vary.